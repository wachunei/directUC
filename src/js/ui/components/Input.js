import styled from "styled-components";

const Input = styled.input`
  border: 1px solid ${(props) => props.theme.colors.inputBorder};
  background: ${(props) => props.theme.colors.inputBackground};
  padding: 0.5rem;
  border-radius: 0;
  width: ${(props) => (props.block ? "100%" : "unset")};

  &:focus {
    border: 1px solid ${(props) => props.theme.colors.primary};
  }

  &:disabled {
    cursor: not-allowed;
    filter: grayscale(100%) opacity(70%);
  }
`;

export default Input;
