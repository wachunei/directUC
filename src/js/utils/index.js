export const removeSuffix = (suffix, value) =>
  value.endsWith(suffix) ? value.slice(0, -suffix.length) : value;

export const parseDistinguishedName = (name) =>
  name.replace(
    /\w\S*/g,
    (txt) => txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase()
  );

export const isFirefox = !window.chrome.app;
