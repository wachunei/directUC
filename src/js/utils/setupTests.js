// eslint-disable-next-line import/no-extraneous-dependencies
import "jest-styled-components";
