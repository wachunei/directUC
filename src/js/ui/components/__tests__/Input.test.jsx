import React from "react";
import Input from "../Input";
import { renderTree } from "./utils";
import "jest-styled-components";

describe("Input Component", () => {
  test("should match snapshot", () => {
    const tree = renderTree(<Input />);
    expect(tree).toMatchSnapshot();
  });

  test("should accept props", () => {
    const tree = renderTree(<Input type="text" value="Input value" />);
    expect(tree).toMatchSnapshot();
  });
});
