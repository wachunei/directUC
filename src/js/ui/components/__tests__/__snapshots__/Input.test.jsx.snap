// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component should accept props 1`] = `
.c0 {
  border: 1px solid #909090;
  background: #FFFFFF;
  padding: 0.5rem;
  border-radius: 0;
  width: unset;
}

.c0:focus {
  border: 1px solid #1F95EB;
}

.c0:disabled {
  cursor: not-allowed;
  -webkit-filter: grayscale(100%) opacity(70%);
  filter: grayscale(100%) opacity(70%);
}

<input
  className="c0"
  type="text"
  value="Input value"
/>
`;

exports[`Input Component should match snapshot 1`] = `
.c0 {
  border: 1px solid #909090;
  background: #FFFFFF;
  padding: 0.5rem;
  border-radius: 0;
  width: unset;
}

.c0:focus {
  border: 1px solid #1F95EB;
}

.c0:disabled {
  cursor: not-allowed;
  -webkit-filter: grayscale(100%) opacity(70%);
  filter: grayscale(100%) opacity(70%);
}

<input
  className="c0"
/>
`;
